### PR TITLE
Add support for subsystem configuration for baremetal use cases

### PR DESCRIPTION
**Pull request by kedareswararao**
Add support for subsystem configuration for baremetal use cases...
@zeddii , When you have sometime please review the pull request and let me know the review comments...

Regards,
Kedar.

### CODE REVIEW

**Comment by zeddii** [comment]
I'll have a look shortly. I'm just finishing up some other urgent fixes and then will review.

File: assists/baremetalconfig_xlnx.py [hunk 1]
```diff
@@ -309,7 +309,48 @@ def get_mapped_nodes(sdt, node_list, options):
...
+        try:
+            is_subsystem = node["cpus"].value
+            if is_subsystem:
+                domain_nodes.append(node)
+        except:
+            pass
+
+    if domain_nodes:
+        # Get Matched Domains for a given cpu cluster
+        cpu_phandle = Lopper.node_getphandle(sdt.FDT, match_cpunodes[0].parent.number)
```
Comment (zeddii) [inline assists/baremetalconfig_xlnx.py:327]: While this is ok, there are some incoming features that further split the lopper parser and backend from libfdt, including the introducing of a dtc-free run.  The Lopper class will contain the 'always available' methods, while front/backend specific ones will move to subclasses (and in this example, one for the FDT). 

If an assist is to work with all font/backends, it shouldn't directly reference the FDT (but instead, use the LopperTree routines).

In this case, you could simply access the parent.phandle, and if it is non-zero, you know the node has a valid phandle.

**Comment by zeddii** [comment]
One comment on the use of the FDT, but otherwise, this looks fine.

File: assists/baremetalconfig_xlnx.py (hunk 1 above)
Comment (kedareswararao) [reply assists/baremetalconfig_xlnx.py:327]: Thanks for the review... 
will fix and send the updated pull request...

**Comment by kedareswararao** [comment]
@zeddii , Updated assist as per your review comments and force pushed the change, Please let me know if I need to send new pull request will close this and will send new one...

**Comment by zeddii** [comment]
This same request is fine. I'll have a quick look today, and will get the merge done.

**Comment by zeddii** [comment]
merged.

**Pull request closed without merging**